airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -103,6 +106,7 @@ def __init__(
         self,
         *,
         dynamodb_table_name: str,
+        dynamodb_conn_id: str | None = None,
```

Comment:
It is more about less known behaviour of boto3-based Hooks, `None` it is a legit value and could force to use boto3 default behaviour strategy (without any call to Connections), e.g. env Variables, IAM Profile, ECS Task Role and etc.

I think better to use specific sentinel as default value instead of `None` here

https://github.com/apache/airflow/blob/a76e0fe16ef12749c3fea1b68d82936b238fafbb/airflow/utils/types.py#L28-L44

Reply:
Thanks for this feedback, @Taragolis, it's very much appreciated as I'm quite new to the Airflow codebase.

Reply:
Well this behaviour not well documented - only in docstrings mentioned that `aws_conn_id` (which propagated to AwsBaseHook) could apply `None` which internally fallback to simple `boto3.session.Session(region_name=region_name)`. We add some info about other very odd historical behaviour in  documentation for connection (see main brach [docs](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow-providers-amazon/latest/connections/aws.html#default-connection-ids)) which also mention about `None` behaviour.

In Amazon Provider still a lot of inconsistent things between components, especially in Operators and hooks which designed in Airflow 1.x, however every release situation become better and better.